dart/math/MathTypes.h

```diff
@@ -83,8 +85,28 @@ typedef Matrix<double, 6, 6> Matrix6d;
 typedef std::vector<Eigen::Vector3d> EIGEN_V_VEC3D;
 typedef std::vector<std::vector<Eigen::Vector3d > > EIGEN_VV_VEC3D;
 
+template <typename _Tp>
+using aligned_vector = std::vector<_Tp, Eigen::aligned_allocator<_Tp>>;
+
+template <typename _Key, typename _Tp, typename _Compare = std::less<_Key>>
+using aligned_map = std::map<_Key, _Tp, _Compare,
+    Eigen::aligned_allocator<std::pair<const _Key, _Tp>>>;
+
+template<typename _Tp, typename... _Args>
+inline std::shared_ptr<_Tp> make_aligned_shared()
```

Comment:
Actually, I thought that too. So I double checked it with clang on linux and mac because clang seems more compatible than others to me. Once I observed that clang also complains similarly, I started to investigate on Eigen.

Beside on the compilation issue, I confirmed that the memory alignment issue is also resolved on Ubuntu 32-bit 14.04. So I believe this PR is ready to merge.